doctests.py

```diff
@@ -104,27 +104,36 @@ def __init__(self, return_code, output):
     def result_text(self):
         return "ERROR"
 
-
-def run_test(args):
+def check_test(proc):
     try:
-        check_output(args, stderr=STDOUT, timeout=5)
-    except CalledProcessError as e:
-        return CommandError(e.returncode, e.output.decode('utf-8'))
+        out, err = proc.communicate(timeout=5)
+        if proc.returncode != 0:
+            return CommandError(proc.returncode, out)
     except TimeoutExpired:
+        proc.kill()
         return Timeout()
```

Comment:
Would it be possible to return output when there's a timeout to indicate how far the test had got?

Reply:
I guess you could make the timeout class a bit more like CommandError... then insert the output to see where it got stuck
I'll have a look tomorrow

Reply:
I've tweaked the classes a bit.

Reply:
I don't think the change works, `out` won't have been initialised as it is `proc.communicate(timeout=5)` that throws.
Having a quick look locally [TimoutExpired has an `.output`](https://docs.python.org/3/library/subprocess.html#subprocess.TimeoutExpired.output), but it is `None` due to how it is being called.

Do you want to just revert / rebase and drop ffaac69 ?

Reply:
Fair enough. That's a shame would've been nice if it worked.. Might have to wait till tomorrow....

Reply:
Have reverted and am getting an error in the pipeline - TIMEOUT apparently

Reply:
The irony 😆 
> README.md:83: python ./run-wheatley 238915467 --method "Plain Bob Major"
TIMED OUT
❌ Documentation tests failed
Error: Process completed with exit code 1.

It occurred to me the timeout only really applies to the first test.
All the other tests get the total time for preceding tests + 5s.

Reply:
Perhaps we should just increase it to 10s @kneasle?

Reply:
Yep I think 10s is better - after all we are running all the doctests in parallel so they're bound to be slightly slower than they were before.